lambda/lambda.ml

```diff
@@ -879,7 +881,41 @@ type lambda =
   | Lexclave of lambda
   | Lsplice of lambda_splice
 
-and slambda = lambda SL.t0
+and slambda =
+  | SLquote of lambda
+  | SLlayout of layout
+  | SLvar of Ident.t
+  | SLrecord of slambda Ident.Map.t
+  | SLfield of Ident.t * Ident.t
```

Comment:
I recall we discussed the use of `Ident.t` for record labels, and don't recall what our exact conclusion was. My instinct, though, is to prefer `string`, as typed tree does.

I guess the reason to want to use `Ident` is that we're going to build `slambda` records corresponding to modules, and the fields of modules are named with idents. But we're also going to want to build `slambda` records correponding to records, where the fields are strings.  And I actually think that, in the module case, just using arrays would be fine - transl already translates module accesses into projections by field number.

Reply:
Well, I certainly agree that we can't just take the name part of an ident, but I think it should be straightforward to convert an ident to a unique string (since they have disambiguating stamps). And converting idents to strings seems better than converting strings to idents, if we want to just have one type for static products. Having multiple types is certainly an option too, though.